tests/mowgli_etl_test/pipeline/wdc/offers_corpus_english_v2_random_sample_test.py

```diff
@@ -1,19 +1,22 @@
 from mowgli_etl.pipeline.wdc.wdc_offers_corpus import WdcOffersCorpus
 from pathlib import Path
+import pytest
 
-import warnings
-warnings.simplefilter(action='ignore', category=StopIteration)
 
 def test_100_data_sample(wdc_large_offers_corpus: WdcOffersCorpus):
-	count = 0
-	for entry in wdc_large_offers_corpus.sample(100):
-		count += 1
-	assert count == 100
+    count = 0
+    for entry in wdc_large_offers_corpus.sample(100):
```

Comment:
I looked through itertools for something like this. (I'd usually do assert len(tuple(...generator...)) == whatever). No joy, but the rest of that module may be useful to you at some point.